apis/Google.Cloud.Diagnostics.Common/Google.Cloud.Diagnostics.Common/RpcRetryConsumer.cs

```diff
@@ -72,7 +72,10 @@ internal class RpcRetryConsumer<T> : IConsumer<T>
             _timerFactory = GaxPreconditions.CheckNotNull(timerFactory, nameof(timerFactory));
 
             // Start the background timer that will retry failed requests.
-            _timer = _timerFactory(FlushBuffer, _options);
+            if (options.RetryType != RetryType.None)
+            {
```

Comment:
A user will always have to go through this channel.  I'd rather fail here than later if possible.